src/OpenSSL/crypto.py

```diff
@@ -168,12 +168,35 @@ def _set_asn1_time(boundary: Any, when: bytes) -> None:
     """
     if not isinstance(when, bytes):
         raise TypeError("when must be a byte string")
+    if boundary == _ffi.NULL:
+        # ASN1_TIME_set_string validates the string without writing anything
+        # when the destination is NULL.
+        raise ValueError("boundary must not be NULL")
```

Comment:
Is it possible to write a test case that reaches this?

Reply:
Replaced with assert per out-of-band conversation. (This is unreachable. It's just to avoid another such bug being masked.)